### DISCUSSION TITLE
server.close equivalent in koajs?

### DISCUSSION THREAD

**Discussion by ziyaaktas** [koajs/koa#1715] [General]
Does server/app.close exist in koajs or no more?

**Comment by zautumnz** [comment]
`app.listen` [returns a `server.listen`](https://github.com/koajs/koa/blob/master/lib/application.js#L97) which in turn has the same [`close` method](https://nodejs.org/api/http.html#serverclosecallback) as most other Node server libraries/frameworks. This can be tested like so:

```javascript
const Koa = require('koa')
const app = new Koa()
const server = app.listen(8000, () => { console.log('listening') })
setTimeout(() => { server.close(() => { console.log('closing') }) }, 1000)
```

**Comment by jonathanong** [comment]
`new Koa().listen()` is just a convenience method. In my apps, I always just create the http server myself:

```js
const http = require('http')
const Koa = require('koa')

const app = new Koa()

const fn = app.callback()

const server = http.createServer(fn)

server.listen()

server.close()
```
